common/djangoapps/enrollment/views.py

```diff
@@ -670,3 +677,39 @@ def post(self, request):
                     actual_activation=current_enrollment['is_active'] if current_enrollment else None,
                     user_id=user.id
                 )
+
+
+@can_disable_rate_limit
+class BulkEnrollmentView(APIView):
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        user = request.user
+        course_ids = request.data.get('course_ids')
+
+        if not course_ids:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={"message": u"Course IDs must be specified to create new enrollments."}
+            )
+
+        # convert course ids to course keys
+        course_keys = []
+        for course_id in course_ids:
+            try:
+                course_keys.append(CourseKey.from_string(course_id))
+            except InvalidKeyError:
+                return Response(
+                    status=status.HTTP_400_BAD_REQUEST,
+                    data={
+                        "message": u"No course '{course_id}' found for enrollment".format(course_id=course_id)
+                    }
+                )
+
+        response = api.add_bulk_enrollments(username=user.username,
+                                            course_ids=course_keys,
+                                            check_access=False,
+                                            request=request)
+
+        return Response(response)
```

Comment:
The problem with this is that the behavior is quite different from EnrollmentListView. Best thing is factor out the common behavior needed and make a call from both functions.